pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1610,6 +1607,16 @@ public void increaseAvailablePermits(int delta) {
         increaseAvailablePermits(cnx(), delta);
     }
 
+    public void setMaxReceiverQueueSize(int newSize) {
```

Comment:
Do you think it'd be worth verifying that the `newSize` is positive? I see that the previous implementation did not verify the input, but a non-positive value will completely halt the consumer. Although, it's possible that someone was using that as a feature to effectively "pause" the consumer. Maybe we need to leave it as is. Let me know what you think.

Reply:
> Do you think it'd be worth verifying that the `newSize` is positive? I see that the previous implementation did not verify the input, but a non-positive value will completely halt the consumer. Although, it's possible that someone was using that as a feature to effectively "pause" the consumer. Maybe we need to leave it as is. Let me know what you think.

Added `checkArgument`. And we already have `pause` method, it's too tricky to do negative queue size, let's not encourage this.

Reply:
> Nit: I think we want to add the `@Override` annotation to this method.

done